.github/container/nsys_jax/nsys_jax/scripts/nsys_jax_combine.py

```diff
@@ -0,0 +1,155 @@
+
+    mirror_dir = pathlib.Path(tempfile.mkdtemp()) if len(args.analysis) else None
+    with zipfile.ZipFile(args.output, "w") as ofile:
+        for n_input, input in enumerate(args.input):
+            first_input = n_input == 0
+            keep_this_nsys_rep = args.keep_nsys_rep == "all" or (
+                args.keep_nsys_rep == "first" and first_input
+            )
+            with zipfile.ZipFile(input) as ifile:
+                for member in ifile.infolist():
+                    if member.is_dir():
+                        continue
+                    filename = member.filename
+                    assert filename in hashes
+                    seen_hashes = hashes[filename]
+
+                    def write(dst_info):
+                        assert dst_info.filename not in set(ofile.namelist())
+                        with ifile.open(member) as src:
+                            with ofile.open(dst_info, "w") as dst:
+                                shutil.copyfileobj(src, dst)
+                            if mirror_dir is not None:
+                                dst_path = mirror_dir / dst_info.filename
+                                os.makedirs(dst_path.parent, exist_ok=True)
+                                src.seek(0)
+                                with open(dst_path, "wb") as dst:
+                                    shutil.copyfileobj(src, dst)
+
+                    if filename.endswith(".nsys-rep"):
+                        assert len(seen_hashes) == 1
+                        if filename == input.stem + ".nsys-rep" and keep_this_nsys_rep:
+                            # `filename`` is the .nsys-rep from `input``
+                            write(member)
+                    else:
+                        if len(seen_hashes) == 1:
+                            # This file was the same in all inputs: copy it once.
+                            if first_input:
+                                write(member)
+                        else:
+                            # This file was not the same in all inputs: copy it to a
+                            # modified destination. An input file A/B in reportN.zip will
+                            # be saved as A/B/reportN in the output, i.e. A/B will be a
+                            # directory instead of a file. TODO: in future instead of using
+                            # input.stem use a standardised format showing the device
+                            # numbers that were profiled in reportN.zip.
+                            dst_info = copy.copy(member)
+                            dst_info.filename = filename + "/" + input.stem
+                            write(dst_info)
+        if len(args.analysis):
+            assert mirror_dir is not None
+            # Execute post-processing recipes and add any outputs to `ofile`
+            for analysis in args.analysis:
+                result, output_prefix = execute_analysis_script(
```

Comment:
FWIW this has been factored out and shared between `nsys-jax` and `nsys-jax-combine`